src/mock.rs

```diff
@@ -0,0 +1,151 @@
+use crate::expectation::model::{Expectation, HttpMethod};
+
+pub struct IncomingRequest {
+    pub method: HttpMethod,
+    pub code: i32,
+    pub path: String,
+}
+
+pub fn look_for_expectation(
+    expectations: &Vec<Expectation>,
+    req: IncomingRequest,
+) -> Option<&Expectation> {
+    expectations.iter().find(|e| {
+        let match_method = e.request.method.map(|m| m == req.method).unwrap_or(true);
```

Comment:
To avoid `Copy` you can ask a ref for `HttpMethod` from e using this code : `e.request.method.as_ref()`and then make a comparison through references :
`e.request.method.as_ref().map(|m| m == &req.method)`